wait-for.c

```diff
@@ -270,15 +268,18 @@ int main(int argc, const char **argv) {
 		goto exit;
 	}
 
+	dirpath_buffer = strdup(extrav[0]);
+	if (!dirpath_buffer) {
+		perror("could not create working buffer for given path");
```

Comment:
There's no guarantee that `perror` doesn't peform an allocation, so you should instead just write "error: out of memory" since that's the only reason why strdup would fail

Reply:
Yes, true - but there are cases where both would fail, too. Further, your case is much less likely since the shell prevents arguments being too long in most modern cases.

Also, that error message doesn't make a whole lot of sense to the user, whereas `out of memory` leaves no room for doubt.